test/stringio/test_stringio.rb

```diff
@@ -922,6 +922,16 @@ def test_encoding_write
     assert_equal("abc".encode("utf-32be"), s.string)
   end
 
+  def test_write_str_modify
+    seek_pos = [0, 2, 4, 6]
+    seek_pos.each do |pos|
```

Comment:
```suggestion
    seek_poses = [0, 2, 4, 6]
    seek_poses.each do |pos|
```

BTW, do we need to check multiple positions?

Reply:
For testing this bug, `pos = 2` is enough.
0, 2, 4, 6 corresponds to these cases and I think it's worth covering them.
```
0: no extend, the `else` branch removed in this pull request
"XXXd"

2: extend
"abXXX"

4: append, `if (ptr->pos == olen)` branch in strio_write
"abcdXXX"

6: extend with \0 padding, `if (pos > olen)` branch in strio_extend
"abcd00XXX"
```

Reply:
I see. Could you split the current test to these cases for easy to understand something like the following?

```ruby
def test_write_code_range_no_extend
end

def test_write_code_range_extend
end

def test_write_code_range_append
end

def test_write_code_range_extend_with_0_padding
end
```